projects/pt1/python/torch_mlir_e2e_test/test_suite/elementwise.py

```diff
@@ -2700,10 +2700,10 @@ def forward(self):
         return torch.add(self.x, 3)
 
 
-@register_test_case(
-    module_factory=lambda: ElementwiseAddScalar_TensorLiteralInt32_Module())
-def ElementwiseAddScalar_TensorLiteralInt32_Module_basic(module, tu: TestUtils):
-    module.forward()
+# @register_test_case(
```

Comment:
dead code?

Reply:
Thanks for review!
I am not sure how to fix it properly, this test causes segmentation fault, but it enabled for STABLEHLO and TOSA. 
So i need some guidance with this case. I can't mark as xfailed, as it segfault, but it looks like memory issue. 
https://github.com/llvm/torch-mlir/blob/a8538e1e3fb98c5d2c6170dd0e5d549d1d6b2632/projects/pt1/e2e_testing/xfail_sets.py#L514
https://github.com/llvm/torch-mlir/blob/a8538e1e3fb98c5d2c6170dd0e5d549d1d6b2632/projects/pt1/e2e_testing/xfail_sets.py#L1040

Also, I believe that disabling the test is not recommended, so I just wanted to highlight the issue.

Reply:
Yeah seems like an important test. Can you post the stack trace of the segfault with gdb?

Reply:
Sure, 
build: 
```cmake
cmake -GNinja -Bbuild   -DCMAKE_BUILD_TYPE=Debug   -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++  -DPython3_FIND_VIRTUALENV=ONLY   -DLLVM_ENABLE_PROJECTS=mlir   -DLLVM_EXTERNAL_PROJECTS="torch-mlir"   -DLLVM_EXTERNAL_TORCH_MLIR_SOURCE_DIR="$PWD"   -DMLIR_ENABLE_BINDINGS_PYTHON=ON   -DLLVM_TARGETS_TO_BUILD=host   externals/llvm-project/llvm
cmake --build build
```

just example of failure:
```
python -m e2e_testing.main -c "torchdynamo" -f "ElementwiseAddScalar_TensorLiteralInt32_Module_basic" -v -s
TORCH_VERSION_FOR_COMPARISON = 2.2.0.dev20231105
Compiling ElementwiseAddScalar_TensorLiteralInt32_Module_basic...
Running ElementwiseAddScalar_TensorLiteralInt32_Module_basic...
Segmentation fault (core dumped)
```

Backtrace:
```gdb
gdb --args python -m e2e_testing.main -c "torchdynamo" -f "ElementwiseAddScalar_TensorLiteralInt32_Module_basic" -v -s

(gdb) bt
#0  0x00007ffff7fb51ac in _mlir__mlir_ciface_ElementwiseAddScalar_TensorLiteralInt32_Module ()
#1  0x00007ffff75d1a4a in ffi_call_unix64 ()
   from /localdisk/dmitriim/miniconda3/envs/test_static/lib/python3.11/lib-dynload/../../libffi.so.8
#2  0x00007ffff75d0fea in ffi_call_int () from /localdisk/dmitriim/miniconda3/envs/test_static/lib/python3.11/lib-dynload/../../libffi.so.8
#3  0x00007ffff75e029a in _call_function_pointer (argtypecount=<optimized out>, argcount=1, resmem=0x7fffffffba80, restype=<optimized out>,
    atypes=<optimized out>, avalues=<optimized out>,
    pProc=0x7ffff7fb51a0 <_mlir__mlir_ciface_ElementwiseAddScalar_TensorLiteralInt32_Module>, flags=4353)
    at /usr/local/src/conda/python-3.11.0/Modules/_ctypes/callproc.c:918
#4  _ctypes_callproc (pProc=0x7ffff7fb51a0 <_mlir__mlir_ciface_ElementwiseAddScalar_TensorLiteralInt32_Module>, argtuple=<optimized out>,
    flags=4353, argtypes=<optimized out>, restype=0x555555a964c0 <_Py_NoneStruct>, checker=0x0)
    at /usr/local/src/conda/python-3.11.0/Modules/_ctypes/callproc.c:1254
#5  0x00007ffff75e92ce in PyCFuncPtr_call (self=<optimized out>, inargs=<optimized out>, kwds=0x0)
    at /usr/local/src/conda/python-3.11.0/Modules/_ctypes/_ctypes.c:4181
#6  0x00005555557306b4 in _PyObject_MakeTpCall (tstate=0x555555ad3d98 <_PyRuntime+166328>, callable=0x7fff2d204d50, args=<optimized out>,
    nargs=<optimized out>, keywords=0x0) at /usr/local/src/conda/python-3.11.0/Objects/call.c:214
#7  0x000055555573cc62 in _PyEval_EvalFrameDefault (tstate=<optimized out>, frame=<optimized out>, throwflag=<optimized out>)
    at /usr/local/src/conda/python-3.11.0/Python/ceval.c:4772
#8  0x0000555555783d2d in _PyEval_EvalFrame (throwflag=0, frame=0x7ffff7fb6700, tstate=0x555555ad3d98 <_PyRuntime+166328>)
    at /usr/local/src/conda/python-3.11.0/Include/internal/pycore_ceval.h:73
#9  _PyEval_Vector (kwnames=<optimized out>, argcount=<optimized out>, args=<optimized out>, locals=0x0, func=0x7fff2eebb7e0,
    tstate=0x555555ad3d98 <_PyRuntime+166328>) at /usr/local/src/conda/python-3.11.0/Python/ceval.c:6428
#10 _PyFunction_Vectorcall (kwnames=<optimized out>, nargsf=<optimized out>, stack=<optimized out>, func=0x7fff2eebb7e0)
    at /usr/local/src/conda/python-3.11.0/Objects/call.c:393
#11 _PyObject_VectorcallTstate (tstate=0x555555ad3d98 <_PyRuntime+166328>, callable=0x7fff2eebb7e0, args=<optimized out>,
    nargsf=<optimized out>, kwnames=<optimized out>) at /usr/local/src/conda/python-3.11.0/Include/internal/pycore_call.h:92
#12 0x0000555555783315 in method_vectorcall (method=<optimized out>, args=0x7fff2d6b1948, nargsf=<optimized out>, kwnames=0x0)
    at /usr/local/src/conda/python-3.11.0/Objects/classobject.c:89
#13 0x0000555555740a45 in do_call_core (use_tracing=<optimized out>, kwdict=0x0, callargs=0x7fff2d6b1930, func=0x7fff2d214f00,
    tstate=<optimized out>) at /usr/local/src/conda/python-3.11.0/Python/ceval.c:7350
#14 _PyEval_EvalFrameDefault (tstate=<optimized out>, frame=<optimized out>, throwflag=<optimized out>)
    at /usr/local/src/conda/python-3.11.0/Python/ceval.c:5377
#15 0x0000555555764113 in _PyEval_EvalFrame (throwflag=0, frame=0x7ffff7fb6638, tstate=0x555555ad3d98 <_PyRuntime+166328>)
    at /usr/local/src/conda/python-3.11.0/Include/internal/pycore_ceval.h:73
#16 _PyEval_Vector (kwnames=<optimized out>, argcount=0, args=0x555555ab9a10 <_PyRuntime+58928>, locals=0x0, func=<optimized out>,
    tstate=0x555555ad3d98 <_PyRuntime+166328>) at /usr/local/src/conda/python-3.11.0/Python/ceval.c:6428
#17 _PyFunction_Vectorcall (func=<optimized out>, stack=0x555555ab9a10 <_PyRuntime+58928>, nargsf=<optimized out>, kwnames=<optimized out>)
    at /usr/local/src/conda/python-3.11.0/Objects/call.c:393
#18 0x0000555555740a45 in do_call_core (use_tracing=<optimized out>, kwdict=0x0, callargs=0x555555ab99f8 <_PyRuntime+58904>,
    func=0x7fff2d515760, tstate=<optimized out>) at /usr/local/src/conda/python-3.11.0/Python/ceval.c:7350
#19 _PyEval_EvalFrameDefault (tstate=<optimized out>, frame=<optimized out>, throwflag=<optimized out>)
    at /usr/local/src/conda/python-3.11.0/Python/ceval.c:5377
#20 0x00005555557fa05e in _PyEval_EvalFrame (throwflag=0, frame=0x7ffff7fb61b8, tstate=0x555555ad3d98 <_PyRuntime+166328>)
--Type <RET> for more, q to quit, c to continue without paging--
    at /usr/local/src/conda/python-3.11.0/Include/internal/pycore_ceval.h:73
#21 _PyEval_Vector (tstate=0x555555ad3d98 <_PyRuntime+166328>, func=0x7ffff7976020, locals=0x7ffff7996840, args=<optimized out>,
    argcount=<optimized out>, kwnames=<optimized out>) at /usr/local/src/conda/python-3.11.0/Python/ceval.c:6428
#22 0x00005555557f95ef in PyEval_EvalCode (co=<optimized out>, globals=0x7ffff7996840, locals=<optimized out>)
    at /usr/local/src/conda/python-3.11.0/Python/ceval.c:1154
#23 0x0000555555814b5e in builtin_exec_impl (module=<optimized out>, closure=<optimized out>, locals=0x7ffff7996840, globals=0x7ffff7996840,
    source=0x7ffff7764ce0) at /usr/local/src/conda/python-3.11.0/Python/bltinmodule.c:1075
#24 builtin_exec (module=<optimized out>, args=<optimized out>, nargs=<optimized out>, kwnames=<optimized out>)
    at /usr/local/src/conda/python-3.11.0/Python/clinic/bltinmodule.c.h:465
#25 0x0000555555749faf in cfunction_vectorcall_FASTCALL_KEYWORDS (func=0x7ffff792cfe0, args=0x7ffff7fb6180, nargsf=<optimized out>,
    kwnames=0x0) at /usr/local/src/conda/python-3.11.0/Include/cpython/methodobject.h:52
#26 0x0000555555749e91 in _PyObject_VectorcallTstate (kwnames=<optimized out>, nargsf=<optimized out>, args=<optimized out>,
    callable=0x7ffff792cfe0, tstate=0x555555ad3d98 <_PyRuntime+166328>)
    at /usr/local/src/conda/python-3.11.0/Include/internal/pycore_call.h:92
#27 PyObject_Vectorcall (callable=0x7ffff792cfe0, args=<optimized out>, nargsf=<optimized out>, kwnames=<optimized out>)
    at /usr/local/src/conda/python-3.11.0/Objects/call.c:299
#28 0x000055555573cc62 in _PyEval_EvalFrameDefault (tstate=<optimized out>, frame=<optimized out>, throwflag=<optimized out>)
    at /usr/local/src/conda/python-3.11.0/Python/ceval.c:4772
#29 0x0000555555764113 in _PyEval_EvalFrame (throwflag=0, frame=0x7ffff7fb6020, tstate=0x555555ad3d98 <_PyRuntime+166328>)
    at /usr/local/src/conda/python-3.11.0/Include/internal/pycore_ceval.h:73
#30 _PyEval_Vector (kwnames=<optimized out>, argcount=2, args=0x7ffff777c798, locals=0x0, func=<optimized out>,
    tstate=0x555555ad3d98 <_PyRuntime+166328>) at /usr/local/src/conda/python-3.11.0/Python/ceval.c:6428
#31 _PyFunction_Vectorcall (func=<optimized out>, stack=0x7ffff777c798, nargsf=<optimized out>, kwnames=<optimized out>)
    at /usr/local/src/conda/python-3.11.0/Objects/call.c:393
#32 0x0000555555826ffb in pymain_run_module (modname=<optimized out>, set_argv0=1) at /usr/local/src/conda/python-3.11.0/Modules/main.c:300
#33 0x00005555558269ca in pymain_run_python (exitcode=0x7fffffffc690) at /usr/local/src/conda/python-3.11.0/Modules/main.c:595
#34 Py_RunMain () at /usr/local/src/conda/python-3.11.0/Modules/main.c:680
#35 0x00005555557e7e19 in Py_BytesMain (argc=<optimized out>, argv=<optimized out>) at /usr/local/src/conda/python-3.11.0/Modules/main.c:734
#36 0x00007ffff7cb9d90 in __libc_start_call_main (main=main@entry=0x5555557e7d70 <main>, argc=argc@entry=9, argv=argv@entry=0x7fffffffc8e8)
    at ../sysdeps/nptl/libc_start_call_main.h:58
#37 0x00007ffff7cb9e40 in __libc_start_main_impl (main=0x5555557e7d70 <main>, argc=9, argv=0x7fffffffc8e8, init=<optimized out>,
    fini=<optimized out>, rtld_fini=<optimized out>, stack_end=0x7fffffffc8d8) at ../csu/libc-start.c:392
#38 0x00005555557e7cb1 in _start ()
```

It looks not so descriptive, so if you have any hints, please share.

Reply:
Ping

Reply:
Ping +1

Reply:
Resolving, as test marked failed.